### DISCUSSION TITLE
Two Factor Authentication (2FA) for peers

### DISCUSSION THREAD

**Discussion by Beoceanq** [WGDashboard/WGDashboard#507] [General]
It would be nice to implement **2FA** for user, any thoughts on this?

**Comment by NoahDar** [comment]
By design wireguard in it's basic form does not support 2FA or any kind of authentication.  You'd need a specialized client with that feature.  

Not saying Donald wouldn't create a client but that would require more coding on both the client and server which I don't think he'd commit that kind of resources to do it.  I'm just happy there is a fantastic WebGUI on the server side that he created and continues to get better.

**Comment by donaldzou** [comment]
Would you like to elaborate on **2FA for user** please? Currently we do have 2FA for admin :)

**Comment by Beoceanq** [comment]
Creating web-pages for authentication, like web server Nginx or Apache, or on VPN server host like 2Fa for admin and create simple authentication page that checks for 2FA (eg Google Authenticator, OTP, or SMS code).
After successful authentication, the web application should record the user's status or give a file for download for further verification by the firewall.
In my mind it will be good, but in design it`s can be very hard)

**Reply by donaldzou** [reply]
Ah gotcha.. you mean a dashboard just for clients.. let me think about that
